### ISSUE TITLE
Repeat-One should use loop points where supported

### ISSUE THREAD

**Issue by mmontag** [mmontag/chip-player-js#55]
Follow up to #51. Some formats have native support for looping, i.e. NSF files that run forever, or MIDI with loop points.

Repeat-One mode should take advantage of this, i.e. Repeat-One should have multiple effects:

- Repeating
    - Lock the sequencer to the current song (advancing the sequencer restarts the song; already implemented)
- Looping
    - Disable auto-advance when player position reaches end of song. Slider handle should stick at the end of the slider range
    - Enable looping behavior within the player engines that support it

These are two distinct behaviors, but I think it makes sense to link them together in the UI.

**Comment by paperluigis** [comment]
yep. i wanted this.

**Comment by paperluigis** [comment]
> Slider handle should stick at the end of the slider range.
No, Like implementing `length / patterns.length * currentPattrIndex` is enough to show the user how NSF / SFC loops.

**Comment by zaphod77** [comment]
repeat one only works for stuff already on the site. not for drag and drop files.

as for looping there should be a "loop count" feature, with a few options to pick from, one of which is infinite, and an option to fade after the loopcount expires.

**Comment by mmontag** [comment]
The existing repeat controls act as high-level sequencer control, and they behave like a traditional music player (Winamp, Spotify).  (Let's ignore the issue of "subtunes" for now...)

Question is: should there be **separate control** for intrinsic looping ("free running" playback)? Programmer brain says yes, it reflects the underlying model honestly. Hmm...

**Comment by zaphod77** [comment]
Looping should respect the loop points in the format, when present. otherwise it should loop from end to start.

It is really hard to do the right thing for mod files in every case, because some of them have subtunes. for example if you play aladdin with "no looping," it will NOT play all of the tunes, it will skip some, because some of them jump forward.

**Comment by mmontag** [comment]
I've pushed a solution for this, trying it out for a while. 
Look for the `[ ] Indefinite Playback` checkbox under the player options.
This will change behavior of looped tracks to continue past the track duration.
Uncheck the box or skip next manually when you get tired of the song.

This is distinct from "Repeat One" because it is not a transport/sequencer behavior (the "CD player buttons"), but a player engine behavior. 

So far it has been implemented for game-music-emu, libvgm, and n64.
